tasks/add_picklist_values.py

```diff
@@ -0,0 +1,339 @@
+    <fields>
+        <fullName>{field}</fullName>
+        <description>{description}</description>
+        <inlineHelpText>{helpText}</inlineHelpText>
+        <label>{label}</label>
+        <type>Picklist</type>
+        <valueSet>
+            <valueSetDefinition>
+                <sorted>{sorted}</sorted>{picklist_values}
+            </valueSetDefinition>
+        </valueSet>
+    </fields>{record_types}
+</CustomObject>"""
+
+picklist_value_template = """
+                <value>
+                    <fullName>{name}</fullName>
+                    <default>{default}</default>
+                    <label>{label}</label>
+                </value>"""
+
+record_type_picklist_template = """
+    <recordTypes>
+        <fullName>{name}</fullName>
+        <active>true</active>
+        <description>{description}</description>
+        <label>{label}</label>
+        <picklistValues>
+            <picklist>{field}</picklist>{picklist_values}
+        </picklistValues>
+    </recordTypes>"""
+
+record_type_picklist_value_template = """
+            <values>
+                <fullName>{name}</fullName>
+                <default>{default}</default>
+            </values>"""
+
+class AddPicklistValues(BaseSalesforceApiTask, Deploy):
+    task_options = {
+        "sobject": {
+            "description": "SObject of the picklist field being modified",
+            "required": True,
+        },
+        "field": {
+            "description": "The picklist field being modified",
+            "required": True,
+        },
+        "values": {
+            "description": "A comma-delimited list of the picklist values being added",
+            "required": True,
+        },
+        "recordtypes": {
+            "description": "The record types for which these new picklist values will be available",
+            "required": False,
+        },
+        "sorted": {
+            "description": "If true, sets the entire picklist to be sorted in alphabetical order",
+            "required": False,
+        },
+        "otherlast": {
+            "description": "If true, the Other value (if one exists) will remain at the end of the values",
+            "required": False,
+        }
+    }
+
+    # Adds picklist values to the given field, if they don't already exist.
+    # Optionally adds the picklist values for the specified record types, if the record types exist.
+    # Optionally updates the picklist values to be alphabetical.
+    def _run_task(self):
+        self.api_version = "47.0"
+        sobject = self.options["sobject"]
+        field = self.options["field"]
+        values = self.options["values"].split(',')
+
+        # The Tooling API does not have the namespace or the '__c' suffix in the DeveloperName of the CustomField object
+        # or in the DeveloperName of the EntityDefinition object, so we have to remove it.
+        split_field = field.split('__')
+        if len(split_field) is 3:
```

Comment:
```suggestion
        if len(split_field) == 3:
```

Always use `==` for comparing values in Python; `is` is for checking to make sure two objects share the same identity (i.e. are the same object). The fact that `is` works here is a bit of a coincidence because Python happens to reuse the same object in memory for small integers, and it's not a property guaranteed by the language.